### DISCUSSION TITLE
Targeting .NET 6

### DISCUSSION THREAD

**Discussion by dclipca** [RageAgainstThePixel/OpenAI-DotNet#395] [Q&A] [answered]
Hi @StephenHodgson. I want to use OpenAI-DotNet in Godot 4.x, However, Godot 4.x officially supports .NET 6, and currently the library doesn't target this framework version.

## Current Situation
Looking at the project files, I notice OpenAI-DotNet currently targets .NET 8.0 and bove. The library would be extremely valuable for people using Godot 4.x, which has standardized on .NET 6 support (LTS).

## Request
Would it be possible to add .NET 6 as a target framework? This would enable usage with Godot 4.x projects and potentially other .NET 6 applications.

## Value Add
- Enables usage in Godot 4.x
- Broadens the library's compatibility
- Helps developers working with local LLM implementations (like KoboldCPP).

.NET 6 is fully backward compatible with newer .NET versions, so this addition (shouldn't?) affect existing functionality.

**Comment by StephenHodgson** [comment, marked as answer]
- https://github.com/RageAgainstThePixel/OpenAI-DotNet/issues/361

.net6 reached end of life and no longer gets support or updates. Unfortunately I did take the breaking changes and incorporated them into the library already and the min supported version is currently .net8.

You could fork this library and add your backwards capabilities but you'd also be losing out on a few performance upgrades.

That being said, I am genuinely surprised Godot doesn't have a way to upgrade the .net target version, and since it is .net6 shouldn't it be fairly trivial to update it since .net6 is no longer in support lifecycle?

**Reply by StephenHodgson** [reply]
> Helps developers working with local LLM implementations (like KoboldCPP).

Also, just to note, I've never fully supported local LLMs and this libraries main goal has always been to support the official sdk for OpenAI and Azure OpenAI.

**Reply by dclipca** [reply]
I agree. It would not make sense to downgrade OpenAI-DotNet to .NET 6 after thinking about it. It seems Godot would upgrade to .NET 8 in the future. Not really familiar with how they implement C# since I use it for a month or so and didn't delve into it. I will think between making a fork and maybe creating another library. It seems .NET doesn't have a library that would unify the API of all those local LLM solutions like KoboldCPP or text-generation-webui. They kind of implement the OpenAI API standard but it's very inconsistent.

**Reply by StephenHodgson** [reply]
Yeah there isn't a good standardization, and everyone seems to want to do it their own slightly different way. Nothing wrong with that but it makes it harder to support everything.

**Reply by dclipca** [reply]
This.

**Reply by StephenHodgson** [reply]
Looks like they just updated 

https://godotengine.org/article/godotsharp-packages-net8/

**Reply by dclipca** [reply]
> Looks like they just updated
> 
> https://godotengine.org/article/godotsharp-packages-net8/

You to try it.
